staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1068,6 +1088,41 @@ func (c *Cacher) LastSyncResourceVersion() (uint64, error) {
 	return c.versioner.ParseResourceVersion(resourceVersion)
 }
 
+// getCurrentResourceVersionFromStorage gets the current resource version from the underlying storage engine.
+// this method issues an empty list request and reads only the ResourceVersion from the object metadata
+func (c *Cacher) getCurrentResourceVersionFromStorage(ctx context.Context) (uint64, error) {
+	if c.newListFunc == nil {
+		return 0, fmt.Errorf("newListFunction wasn't provided for %v", c.objectType)
+	}
+	emptyList := c.newListFunc()
+	pred := storage.SelectionPredicate{
+		Label: labels.Everything(),
+		Field: fields.Everything(),
+		Limit: 1, // just in case we actually hit something
+	}
+	err := c.storage.GetList(ctx, "/reserved_non_existing_prefix", storage.ListOptions{Predicate: pred}, emptyList)
```

Comment:
Hmm - I would actually go with a real key rather then the artifical one.

Reply:
As discussed offline - I don't feel very strongly about that.
Let's keep it as is and let's add a TODO, to reevaluate it before going to Beta.